docs/java-your-first-app.md

```diff
@@ -0,0 +1,163 @@
+
+Next, download the ['java-project-template'](https://github.com/temporalio/java-project-template) by downloading it as a zip or by [creating a new repository](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-from-a-template#creating-a-repository-from-a-template) in your own Github account and then cloning the repo to the location of your choice.
+
+The last setup step is to build the project. If you open it with [IntelliJ](https://www.jetbrains.com/idea/) the project will build automatically. If you prefer to run programs from your terminal, make sure you have [Gradle](https://gradle.org/install/) installed and run the Gradle build command from the root of the project:
+
+```
+./gradlew build
+```
+
+Once your project has built, you are ready to go.
+
+## Application overview
+
+![Tool box](/img/docs/workflow.png)
+
+The Java project template mimics a "money transfer" application in order to give you the minimum elements needed to both get started building your own application as well as understand some of the value that Temporal gives you right out of the box. The project includes a predefined [Workflow](/docs/workflows) that orchestrates the execution of an account object's methods `withdraw()` and `deposit()` to mock the transfer of money from one account to another. Temporal calls such methods [Activities](/docs/activities).
+
+The first thing you will do is to call on the Temporal server to start the transfer. Temporal will then start tracking the progress of the Workflow execution. You will then run a Worker whose sole job is to execute the functions of the Workflow and communicate the results back to the Temporal server.
+
+![High level project design](/img/docs/temporal-high-level-application-design.png)
+
+### The Workflow
+
+With Temporal, when you "start" a Workflow, what you are really doing is telling the Temporal server about code that needs to be run and to start tracking its state. The server then breaks up the code into Tasks and sends each Task to a Task Queue. The code below represents three Tasks, one for the Workflow itself, and one for each of the Activity methods.
+
+<!--SNIPSTART java-project-template-workflow-implementation-->
+<!--SNIPEND-->
+
+### Initiate transfer
+
+Let's tell the server that we want this Workflow code to execute. Make sure the [Temporal server](/docs/install-temporal-server) is running in a terminal, and then from within IntelliJ run the InitiateMoneyTransfer class or from the project root run the following command:
+
+```
+./gradlew initiateTransfer
+```
+
+The two ways to initiate a Workflow with Temporal is either via the SDK or via the CLI. For this tutorial we are focusing on using the SDK to start the Workflow, which is how most Workflows would be initiated in a production environment. Here is the code we just ran to start the Workflow:
+
+<!--SNIPSTART java-project-template-workflow-initiator-->
```

Comment:
How does this actually work? you set workflow id as 'transfer-money-workflow-001' but the implemntation has a different name.

Reply:
"transfer-money-workflow-001" is the "friendly id" of the Workflow. It isn't the actual "ID" used to tell the Worker about it - I don't think? Seems like the identifier that is used to tell the Worker is the class signature which is passed to the server (friendly id is in the options):

```java
 TransferMoneyWorkflow workflow = client.newWorkflowStub(TransferMoneyWorkflow.class, options);
```